### DISCUSSION TITLE
Problem with Dimension in Block

### DISCUSSION THREAD

**Discussion by YanGBloG** [haplokuon/netDxf#342] [Q&A] [answered]
Hi,
When I add a new dimension to a block entity, it's invisible when I open the out .dxf file. And when I edit this block with block editor and save, the dimension becomes visible. Do the issue is a bug or do I need to do something more to make dimension visible without block editor ?
Thank you!

**Comment by haplokuon** [comment, marked as answer]
You are using AutoCad, don't you? Well, you just hit one of its everlasting bugs. You can set the DxfDocument.BuildDimensionBlocks property to true before adding your block to the document. It will create the blocks that make up the graphical representation of the dimension, but keep in mind that the built in dimension constructor in netDxf is limited, it will not use the full range of properties available in the dimension style. Also you might consider to use groups instead, if you can.

**Reply by YanGBloG** [reply]
Yes, I am using AutoCAD, after I set `DxfDocument.BuildDimensionBlocks = true` it works well for me. But when I want to extend the function of dimension, can you tell me where I can get documents to use groups?
And 1 more question, do you have a plan to make PDF from DXF for netDxf in the near future ?

**Reply by haplokuon** [reply]
Groups are just a way to cluster different entities. Take a look at the AutoCad documentation, there, you will find a lot more information about when and how to use them. In netDxf you can create them like this:
```c#
Line line1 = new Line(new Vector2(0, 0), new Vector2(100, 100)){Color = AciColor.Blue};
Line line2 = new Line(new Vector2(100, 0), new Vector2(200, 100)){Color = AciColor.Red};
Line line3 = new Line(new Vector2(200, 0), new Vector2(300, 100)){Color = AciColor.Green};
Line line4 = new Line(new Vector2(300, 0), new Vector2(400, 100)){Color = AciColor.Yellow};

// named group
Group group1 = new Group("MyGroup", new EntityObject[] {line1, line2});

DxfDocument doc = new DxfDocument();
// the Add Group method will also add the entities contained in a group to the document.
doc.Groups.Add(group1);

doc.Entities.Add(line3);
doc.Entities.Add(line4);
//unnamed group with entities already added to the document, they must belong to the same document
Group group2 = new Group(new EntityObject[] {line3, line4});
doc.Groups.Add(group2);

doc.Save("test.dxf");
```
I have no plans to make a PDF writer from the contents of the document in any near future.

**Reply by YanGBloG** [reply]
So, do you mind if I build one from your netDxf ?

**Reply by haplokuon** [reply]
No, not at all. You are free to do as you like, you don't have to ask for my permission. Keep me updated on your progress and if you have any doubts about of the netDxf data structures just ask.
